subprojects/distributions/distributions.gradle.kts

```diff
@@ -34,7 +34,7 @@ tasks.withType<AbstractArchiveTask>().configureEach {
     destinationDirectory.set(rootProject.layout.buildDirectory.dir(rootProject.base.distsDirName))
 }
 
-tasks.named("clean").configure {
+tasks.clean {
     delete(tasks.withType<AbstractArchiveTask>())
```

Comment:
Wow good catch, the receiver of `.configure {}` is `Task` so `delete(..)` was resolving to `Project.delete(...)` instead of `Delete.delete(...)`, effectively doing the deletion when the task was configured instead of at execution.

![image](https://user-images.githubusercontent.com/132773/81475192-e7880200-920a-11ea-88e3-8e4677410a7c.png)

vs

![image](https://user-images.githubusercontent.com/132773/81475223-25852600-920b-11ea-921c-31a5750565a5.png)


This is a trap!

Reply:
Ouch.

Since we have the FileSystemOperations service now, we could consider deprecating the `delete`, `copy` and `sync` methods from `Project` and/or deprecate those things running at configuration time.

Reply:
Yep, deprecating/removing those functions would help alleviate such traps. The same could still be achieved using a plugin with `FileSystemOperations`  injected instead. Using those functions at execution time is already unsupported when instant execution is enabled anyway.

Reply:
Damn! We should have some kind of intercept to catch this.